### PR TITLE
[luci] Introduce SparsityParam

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit introduces SparsityParam.

Related : #3889
Draft : #4190 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Comment by mhs4670go** [comment]
@seanshpark PTAL:)

File: compiler/luci/lang/include/luci/IR/SparsityParam.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  DimMetaData() = delete;
+  DimMetaData(DimensionType format, int32_t dense_size, SparseIndexVectorType array_segments_type,
+              void *array_segments, SparseIndexVectorType array_indices_type, void *array_indices)
+      : _format{format}, _dense_size{dense_size}, _array_segments_type{array_segments_type},
+        _array_segments{array_segments}, _array_indices_type{array_indices_type},
+        _array_indices{array_indices}
+  {
+  }
+
+  DimensionType _format;
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/SparsityParam.h:51]: Please add initial value

File: compiler/luci/lang/include/luci/IR/SparsityParam.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  int32_t _dense_size{0};
+  SparseIndexVectorType _array_segments_type;
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/SparsityParam.h:53]: ditto

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**